### PR TITLE
validate tag policy constrain

### PR DESCRIPTION
**Pull request by StarpTech**
Fixes: #4889

### CODE REVIEW

File: pkg/skaffold/schema/validation/validation.go [hunk 1]
```diff
@@ -56,6 +57,16 @@ func Process(config *latest.SkaffoldConfig) error {
 	return fmt.Errorf(strings.Join(messages, " | "))
 }
 
+// validateTaggingPolicy checks that the tagging policy is valid in combination with other options.
+func validateTaggingPolicy(bc latest.BuildConfig) (errs []error) {
+	if bc.LocalBuild != nil {
+		if bc.LocalBuild.TryImportMissing && bc.TagPolicy.ShaTagger != nil {
+			errs = append(errs, fmt.Errorf("tagging policy 'sha256' can not be used when 'tryImportMissing' is enabled"))
```
Comment (briandealwis) [inline pkg/skaffold/schema/validation/validation.go:64]: We should document why — we may change the sha256:
```suggestion
			// sha256 just uses `latest` tag, so tryImportMissing will virtually always succeed (#4889)
			errs = append(errs, fmt.Errorf("tagging policy 'sha256' can not be used when 'tryImportMissing' is enabled"))
```

File: pkg/skaffold/schema/validation/validation.go [hunk 2]
```diff
@@ -56,6 +57,16 @@ func Process(config *latest.SkaffoldConfig) error {
...
+		}
+	}
+	return
```
Comment (briandealwis) [inline pkg/skaffold/schema/validation/validation.go:67]: Nit: this isn't needed

**Review by briandealwis — COMMENTED** [review]
Thanks for the PR!  Could you please merge with HEAD and address these two issues, and we can get this in?

File: pkg/skaffold/schema/validation/validation.go (hunk 2 above)
Comment (StarpTech) [reply pkg/skaffold/schema/validation/validation.go:67]: This is required because we have a named return value.

**Comment by StarpTech** [comment]
23min for CI is a lot 😄

File: pkg/skaffold/schema/validation/validation.go (hunk 2 above)
Comment (briandealwis) [reply pkg/skaffold/schema/validation/validation.go:71]: Oops! 🤦

**Review by briandealwis — APPROVED** [review]

**Pull request merged**